src/openzaak/components/catalogi/admin/mixins.py

```diff
@@ -122,3 +133,101 @@ def response_change(self, request, obj):
             return HttpResponseRedirect(redirect_url)
 
         return super().response_change(request, obj)
+
+
+class CatalogusImportExportMixin:
+    def import_view(self, request):
+        form = CatalogusImportForm(request.POST, request.FILES)
+        context = dict(self.admin_site.each_context(request), form=form)
+        if "_import" in request.POST:
+            if form.is_valid():
+                try:
+                    import_file = form.cleaned_data["file"]
+                    path = private_media_storage.save(
+                        f"uploads/imports/{import_file.name}",
+                        ContentFile(import_file.read()),
+                    )
+                    call_command(
+                        "import", os.path.join(settings.PRIVATE_MEDIA_ROOT, path)
+                    )
+                    self.message_user(
+                        request,
+                        _("Catalogus successfully imported"),
+                        level=messages.SUCCESS,
+                    )
+                    return HttpResponseRedirect(
+                        reverse("admin:catalogi_catalogus_changelist")
+                    )
+                except CommandError as exc:
+                    self.message_user(request, exc, level=messages.ERROR)
+        return TemplateResponse(
+            request, "admin/catalogi/import_catalogus.html", context
+        )
+
+    def get_related_objects(self, obj):
+        resources = {}
+
+        resources["Catalogus"] = [obj.pk]
+
+        # Resources with foreign keys to catalogus
+        fields = ["InformatieObjectType", "BesluitType", "ZaakType"]
+        for field in fields:
+            resources[field] = list(
+                getattr(obj, f"{field.lower()}_set").values_list("pk", flat=True)
+            )
+        resources["ZaakInformatieobjectType"] = list(
+            ZaakInformatieobjectType.objects.filter(
+                zaaktype__in=resources["ZaakType"],
+                informatieobjecttype__in=resources["InformatieObjectType"],
+            ).values_list("pk", flat=True)
+        )
+
+        # Resources with foreign keys to  ZaakType
+        fields = ["ResultaatType", "RolType", "StatusType", "Eigenschap"]
+        for field in fields:
+            model = apps.get_model("catalogi", field)
+            resources[field] = list(
+                model.objects.filter(zaaktype__in=resources["ZaakType"]).values_list(
+                    "pk", flat=True
+                )
+            )
+
+        resource_list = []
+        id_list = []
+        for resource, ids in resources.items():
+            if ids:
+                resource_list.append([resource])
+                id_list.append([",".join(str(id) for id in ids)])
```

Comment:
can't you pass an actual python list instead of turning it into a comma-separated list? This is passed to the management command args right?